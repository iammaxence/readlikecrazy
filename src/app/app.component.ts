import {Component} from '@angular/core';
import { RouterOutlet } from '@angular/router';
import {TranslateModule} from "@ngx-translate/core";
import {NgForOf} from "@angular/common";

@Component({
  selector: 'app-root',
  standalone: true,
  imports: [RouterOutlet, TranslateModule, NgForOf],
  templateUrl: './app.component.html',
  styleUrl: './app.component.scss',
  providers: []
})
export class AppComponent {}
