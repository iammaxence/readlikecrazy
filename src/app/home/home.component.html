<main class="main container">
  <h1 class="home--title"> Read Like Crazy </h1>
  <div class="home--buttons">
    <button class="home--buttons--item" [id]="menu.title" type="button" *ngFor="let menu of allMenu" (click)="goToPage(menu.routing)"> {{ menu.title | translate }}</button>
  </div>
</main>
