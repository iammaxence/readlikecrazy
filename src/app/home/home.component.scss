.home {
  &--title {
    padding: 5rem ;
    text-align: center;
    font-size: 5rem;
  }
  &--buttons {
    display: flex;
    flex-direction: column;
    gap: 2rem;
    justify-content: center;
    align-items: center;

    &--item {
      width: 10rem;
      height: 3rem;
    }
  }
}
