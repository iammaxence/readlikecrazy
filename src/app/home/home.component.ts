import { Component } from '@angular/core';
import {TranslateModule} from "@ngx-translate/core";
import {NgForOf} from "@angular/common";
import {Router} from "@angular/router";

@Component({
  selector: 'app-home',
  standalone: true,
  imports: [
    TranslateModule,
    NgForOf
  ],
  templateUrl: './home.component.html',
  styleUrl: './home.component.scss'
})
export class HomeComponent {
  allMenu = [
    {
      title: 'menu.words',
      routing: 'read-words',
    },
    {
      title: 'menu.phrases',
      routing: '',
    }
  ]

  constructor(private router: Router) {}

  public goToPage(path: string): void {
    this.router.navigate([path])
  }
}
