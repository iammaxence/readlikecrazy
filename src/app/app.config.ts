import { ApplicationConfig } from '@angular/core';
import { provideRouter } from '@angular/router';

import { routes } from './app.routes';
import {TranslateLoader, TranslateModule} from "@ngx-translate/core";
import {HttpClient, HttpClientModule, provideHttpClient} from "@angular/common/http";
import {TranslateHttpLoader} from "@ngx-translate/http-loader";

export function HttpLoaderFactory(http: HttpClient) {
  return new TranslateHttpLoader(http);
}

export const appConfig: ApplicationConfig = {
  providers: [provideRouter(routes),
    provideHttpClient(),
    TranslateModule.forRoot({
    defaultLanguage: 'en',
    loader: {
      provide: TranslateLoader,
      useFactory: HttpLoaderFactory,
      deps: [HttpClient]
    }
  }).providers!]
};
