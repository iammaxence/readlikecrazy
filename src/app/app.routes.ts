import { Routes } from '@angular/router';
import {ReadWordsComponent} from "./read-words/read-words.component";
import {HomeComponent} from "./home/home.component";
import {Reader} from "./reader/reader";

export const routes: Routes = [
  { path: '', component: HomeComponent },
  { path: 'read-words', component: ReadWordsComponent },
  { path: 'nfive-phrases', component: Reader }
];
