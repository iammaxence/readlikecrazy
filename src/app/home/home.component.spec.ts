import {ComponentFixture, TestBed} from "@angular/core/testing";
import {HomeComponent} from "./home.component";
import {expect} from "vitest";
import {By} from "@angular/platform-browser";
import {
  TranslateLoader,
  TranslateModule,
} from "@ngx-translate/core";
import {HttpLoaderFactory} from "../app.config";
import {HttpClient} from "@angular/common/http";

let component: HomeComponent;
let fixture: ComponentFixture<HomeComponent>;

describe('HomeComponent', () => {
  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [
        HomeComponent,
        TranslateModule.forRoot({
          loader: {
            provide: TranslateLoader,
            useFactory: HttpLoaderFactory,
            deps: [HttpClient]
          }
        })
      ],
      providers: [TranslateLoader],
    }).compileComponents();
    fixture = TestBed.createComponent(HomeComponent);
    component = fixture.componentInstance;
  });

  it('Should init', () => {
    // When
    fixture.detectChanges();

    // Then
    expect(fixture.nativeElement.querySelector('h1').textContent).toContain('Read Like Crazy');
    expect(fixture.nativeElement.querySelectorAll('button').length).toBe(2);
    expect(fixture.debugElement.query(By.css('#menu\\.words')).nativeElement.textContent).toContain('menu.words');
    expect(fixture.debugElement.query(By.css('#menu\\.phrases')).nativeElement.textContent).toContain('menu.phrases');
  });
});
