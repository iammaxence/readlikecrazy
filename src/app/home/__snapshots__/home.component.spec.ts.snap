// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyComponent > should match the snapshot 1`] = `
<app-home>
  <main
    _ngcontent-a-c196787078=""
    class="main container"
  >
    <h1
      _ngcontent-a-c196787078=""
      class="home--title"
    >
       Read Like Crazy 
    </h1>
    <div
      _ngcontent-a-c196787078=""
      class="home--buttons"
    >
      <!--container-->
    </div>
  </main>
</app-home>
`;
